src/kvm/ioctl.rs

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::kvm::types::Handle;
+
+use iocuddle::*;
+
+use std::io::Result;
+use std::marker::PhantomData;
+use std::os::raw::c_uint;
+use std::os::unix::io::AsRawFd;
+
+const REG_REGION: u64 = 0x8010aebb;
+
+pub const ENCRYPTED_REGION: Ioctl<Write, &EncryptedRegion> = unsafe { Ioctl::classic(REG_REGION) };
+
+impl Handle {
+    /// Instruct the kernel to pin these pages so that they aren't remapped
+    /// into a different page frame which will cause unpredictable results
+    /// due to how SEV's keys depend on encrypted regions being static.
+    ///
+    /// It's possible this may become obsolete as there are proposed kernel
+    /// patches to make the kernel pin the pages when they're first touched
+    /// automatically.
+    pub fn pin_pages(&self, kvm_fd: &mut impl AsRawFd, region: &[u8]) -> Result<c_uint> {
+        unsafe {
+            ENCRYPTED_REGION
+                // The kernel accidentally exported this ioctl as _IOR, but it
+                // should have been _IOW.
+                .lie::<Write, &EncryptedRegion>()
```

Comment:
This is the wrong way to use `lie()`. It is `const` for a reason. Use it in the ioctl definition.